### PR TITLE
TF model refactorings

### PR DESCRIPTION
**Pull request by aaarrti**
### Description
- This PR contains refactorings addressing some issue I encountered while working on Relative Stability.

### Implemented changes
  - [x] `TFLeNet` was not a LeNet architecture, updated it, trained it, added weights
  - [x] ` ConvNet1DTF` was using Sequential API not in intended way.
  - [x] In case the model has softmax on top, but we need linear, the model will be rebuilt on each `.predict()` call. 
  Instead I cache the rebuilt models.
  - [x] `TensorFlowModel` was using `__call__` which is not intended for use, unless user needs to retrieve gradients.

### Minimum acceptance criteria
- ???

### CODE REVIEW

File: quantus/helpers/model/tf_model.py [hunk 1]
```diff
@@ -59,33 +111,31 @@ def predict(self, x, **kwargs):
...
 
-        output_layer = Dense(**config)(self.model.layers[-2].output)
-        new_model = Model(inputs=[self.model.input], outputs=[output_layer])
-        new_model.layers[-1].set_weights(weights)
+        if self.softmax and output_activation == activations.linear:
+            logits = self.model.predict(x, **predict_kwargs)
+            return tf.nn.softmax(logits)
 
-        return new_model(x, training=False, **model_predict_kwargs).numpy()
+        # In this case model has a softmax on top, and we want linear.
```
Comment (annahedstroem) [inline quantus/helpers/model/tf_model.py:135]: Thanks for being detail-oriented!

File: quantus/helpers/model/tf_model.py [hunk 2]
```diff
@@ -49,8 +69,40 @@ def __init__(
             softmax=softmax,
             model_predict_kwargs=model_predict_kwargs,
         )
+        # get_hidden_representations needs to rebuild and re-trace the model.
```
Comment (annahedstroem) [inline quantus/helpers/model/tf_model.py:72]: Great explanation!

**Review by annahedstroem — COMMENTED** [review]
Ready for merge!
